app/core/crossref/citation_list.tsx

```diff
@@ -1,76 +1,184 @@
+import { Element, Text } from "xast"
+export interface CitationList extends Element {
+  name: "citation_list"
+  children: Citation[]
+}
```

Comment:
Great question actually haha, I guess that's just what I usually do! To me it's like "here's what we're working with (types) and here's what we do with them (functions)". I also usually prefer to have this "smaller to larger" structure in files, some people like putting functions at the bottom and have them hoisted up, like this:
```ts
export default function xPlusTwoSquared(x: number){
		const xPlusTwo = plusTwo(x)
		return Math.pow(x, 2)
}

function xPlusTwo(x: number){
		return x + 2
}
```

but I usually find that somewhat confusing. When I read `xPlusTwoSquared` and see `xPlusTwo`, I don't know whether that is defined in this file or somewhere else. If you switch them around it's immediately obvious that `xPlusTwo` is defined to be used later, so it's less confusing when you encounter it in `xPlusTwoSquared`. This is also why I like to use `const x = () =>` instead of function declarations, as they force you to be more explicit about the order.

But this is just personal preference and is kind of temporary in this case. With this many types I'd put them in a different file in the end as that makes importing way easier and cleaner, as instead of
```ts
import { Author, Contributors } from './contributors'
import { Book } from './book'
import { Doi } from './doi'
```
You can just do
```ts
import { Author, Contributors, Book, Doi } from './types'
```

Reply:
> Do I understand correctly future work would be to auto generate these types from [the `.xsd` file](https://data.crossref.org/reports/help/schema_doc/5.3.1/index.html)?

And yes, I ideally! The libraries for this are not very well maintained though, I have a [https://github.com/tefkah/cxsd/tree/feat/xast[(fork which creates `unified` compatible types) but it's still not ideal and needs some further adjusting. 

I do want to have a library which does this well though, it's a relatively common usecase for me now, so I think I will clean it up.

I don't think the quality of maintantence is huge concern in this regard, as almost everyone uses XSD v1 which is from 2001